src/main/java/carpet/CarpetSettings.java

```diff
@@ -739,4 +741,47 @@ public static void changeSpawnSize(int size)
             category = {SURVIVAL, CLIENT}
     )
     public static boolean cleanLogs = false;
+
+    @Rule(
+            desc = "Customizable structure block limit of each axis",
+            extra = {"WARNING: Needs to be permanent for correct loading.",
+                    "Setting 'structureBlockIgnored' to air is recommended",
+                    "when saving massive structures.",
+                    "Required on client of player editing the Structure Block.",
+                    "'structureBlockOutlineDistance' may be required for",
+                    "correct rendering of long structures."},
+            options = {"48", "96", "192", "256"},
+            category = {CREATIVE, CLIENT},
+            strict = false
+    )
+    public static int structureBlockLimit = 48;
```

Comment:
I don't think it makes sense for these value to be less than vanilla, meaning I suggest to add validator making sure its at least 48. Not sure about the upper bound limit - because whatever it would be set to - there will always be folk complaining its not as high as they wanted to be. But at least >=48 in this case.